### ISSUE TITLE
Cannot find Message class to DefinedStrategy

### ISSUE THREAD

**Issue by xico002** [connamara/quickfixn#665]
Hi!

I'm trying to receive messages from ICE (This is their website just to inform: https://www.theice.com/index). When using OnMessage() method I cannot find a message that represents DefinedStrategy. Updated my FIX44.xml to request the message 35=c with 321=101. But I need to get a response and for that I need to receive a message that represents the DefinedStrategy message and that class does not exist in QuickFix.Net.NETCore.FIX44 package. 

Can you help me? Am I missing something?

Regards,
Francisco Canhoto

**Comment by gbirchmeier** [comment]
That's a custom message that ICE invented.  You'll need to manually add that to your DataDictionary.

http://quickfixn.org/tutorial/custom-fields-groups-and-messages.html

**Comment by xico002** [comment]
Right but wont that message be rejected if i dont call the OnMessage() method?

**Comment by xico002** [comment]
If quickfix doesnt have a specific class to that message how can I crack the message?

**Comment by gbirchmeier** [comment]
I answered a similar question on the mailing list the other day.

http://lists.quickfixn.com/pipermail/quickfixn-quickfixn.com/2021q1/thread.html

See the thread called "Custom Dictionaries"

**Comment by gbirchmeier** [comment]
@DocSatherley I am checking with our IT guy to figure out where those list archives went.  Thanks for alerting me.

**Comment by DocSatherley** [comment]
@gbirchmeier did your IT guy have any luck locating the archives?

**Comment by gbirchmeier** [comment]
@DocSatherley He has a ticket out with the hosting provider.

But I can probably answer your question here.  What exactly do you need to know?

**Comment by DocSatherley** [comment]
@gbirchmeier I am doing the same thing the original poster was trying to do.  ICE have a custom DefinedStrategy message.  I have created a data dictionary to match the ICE specification.  I have also created a class and OnMesaage method for DefinedStrategy messages (both only scaffolding at the moment).  However, calls to Crack are throwing UnsupportedMessageType.  I can see my method in _callCache when debugging in Visual Studio.  I am assuming I have missed part of the process and was hoping the thread will tell me what.
